purchases/src/main/kotlin/com/revenuecat/purchases/common/Backend.kt

```diff
@@ -441,14 +453,73 @@ internal class Backend(
         synchronized(this@Backend) {
             diagnosticsCallbacks.addCallback(
                 call,
-                diagnosticsDispatcher,
+                eventsDispatcher,
                 cacheKey,
                 onSuccessHandler to onErrorHandler,
                 Delay.LONG,
             )
         }
     }
 
+    @RequiresApi(Build.VERSION_CODES.N)
+    fun postPaywallEvents(
+        paywallEventRequest: PaywallEventRequest,
+        onSuccessHandler: () -> Unit,
+        onErrorHandler: (error: PurchasesError, shouldMarkAsSynced: Boolean) -> Unit,
+    ) {
+        val body = PaywallEventsManager.json.encodeToJsonElement(paywallEventRequest).asMap()
+        if (body == null) {
+            onErrorHandler(
+                PurchasesError(
+                    PurchasesErrorCode.UnknownError,
+                    "Error encoding paywall event request",
+                ).also { errorLog(it) },
+                true,
+            )
+            return
+        }
+        val call = object : Dispatcher.AsyncCall() {
+            override fun call(): HTTPResult {
+                return httpClient.performRequest(
+                    appConfig.paywallEventsURL,
+                    Endpoint.PostPaywallEvents,
```

Comment:
It took me a while to find where this was defined. I feel like `paywallEventsURL` should be part of the `Endpoint` configuration. What do you think? Otherwise it would be possible (at compile time) to make a request to conflicting endpoints / domains.

Reply:
Hmm we could certainly move it there, but I also feel with the upcoming work for adding fallback base URLs, it might be better to keep separate... Honestly, I could see it going either way.

Reply:
I'm gonna hold on this until we actually rewrite the networking code to not increase the size of this PR even more.

Reply:
Sounds good, and that's a good point.